samplers/samplers_test.go

```diff
@@ -418,3 +421,82 @@ func TestMetricKeyEquality(t *testing.T) {
 	assert.Equal(t, ce1.MetricKey.String(), ce2.MetricKey.String())
 	assert.NotEqual(t, ce1.MetricKey.String(), ce3.MetricKey.String())
 }
+
+func TestParseMetricSSF(t *testing.T) {
+	sample := ssf.SSFSample{
+		Metric: ssf.SSFSample_GAUGE,
+
+		Name:       "my.test.metric",
+		Value:      rand.Float32(),
+		Timestamp:  time.Now().Unix(),
+		Message:    "arbitrary test message",
+		Status:     ssf.SSFSample_WARNING,
+		SampleRate: rand.Float32(),
+		Tags: map[string]string{
+			"keats":            "false",
+			"yeats":            "false",
+			"wilde":            "true",
+			"veneurglobalonly": "true",
+		},
+		Unit: "frobs per second",
+	}
+
+	expected := UDPMetric{
+		MetricKey: MetricKey{
+			Name:       "my.test.metric",
+			Type:       "gauge",
+			JoinedTags: "keats:false,wilde:true,yeats:false",
+		},
+		Digest:     0x7ae783ad,
+		Value:      sample.Value,
+		SampleRate: sample.SampleRate,
+		Tags: []string{
+			"keats:false",
+			"wilde:true",
+			"yeats:false",
+		},
+		Scope: 2,
+	}
+
+	udpMetric, err := ParseMetricSSF(&sample)
+	assert.NoError(t, err)
+	assert.Equal(t, udpMetric.MetricKey, expected.MetricKey)
+	assert.Equal(t, udpMetric.Type, expected.Type)
+	assert.Equal(t, udpMetric.Digest, expected.Digest)
+	assert.InEpsilon(t, udpMetric.Value, expected.Value, ε)
+	assert.InEpsilon(t, udpMetric.SampleRate, expected.SampleRate, ε)
+	assert.Equal(t, udpMetric.JoinedTags, expected.JoinedTags)
+	assert.Equal(t, udpMetric.Tags, expected.Tags)
+	assert.Equal(t, udpMetric.Scope, expected.Scope)
+}
+
+func BenchmarkParseMetricSSF(b *testing.B) {
+
+	const LEN = 10000
+
+	samples := make([]*ssf.SSFSample, LEN)
+
+	for i, _ := range samples {
+		p := make([]byte, 10)
+		_, err := rand.Read(p)
+		if err != nil {
+			b.Fatalf("Error generating data: %s", err)
+		}
+		sample := ssf.SSFSample{
+			Name:       "my.test.metric",
+			Value:      rand.Float32(),
+			Timestamp:  time.Now().Unix(),
+			SampleRate: rand.Float32(),
+			Tags: map[string]string{
+				"keats":       "false",
+				"yeats":       "false",
+				"wilde":       "true",
+				string(p[:5]): string(p[5:]),
+			},
+		}
+		samples[i] = &sample
+	}
+	for i := 0; i < b.N; i++ {
```

Comment:
Might not be a bad idea to add in a `b.ResetTimer()` call here, although I bet this initialization isn't all that expensive in the first place.

Reply:
Definitely should, as doing so also resets allocation counts when running with `-benchmem`.